### ISSUE TITLE
Context.hasListener is not working inside another component

### ISSUE THREAD

**Issue by ceremcem** [ractivejs/ractive#3251]
## Description:

I would expect both `mycomp` detect their `hello` handler: 
```html
<mycomp on-hello="there">This has a listener</mycomp>
<something>
  <mycomp on-hello="there">This has a listener</mycomp>
</something>
```
Which outputs:
![image](https://user-images.githubusercontent.com/6639874/41571174-9fdb15ce-737a-11e8-9e04-7b8218170e24.png)

Where the problem is that the one inside `something` component (inside dashed `div`) can not detect its `hello` handler. 
## Versions affected:
0.10.3

## Reproduction 

[Playground](https://ractive.js.org/playground/?env=docs#N4IgFiBcoE5SAbAhgFwKYGcUgL4BoQN4AlJAYxQEsA3NAOjIHsBbAB0YDs0OUM7mAnkzYACALwjSFGvTQAPdBwAmACmAAdDupTo2ydJBEADTdu0AeAEYBXHZxGcAtGQSUyAazHrwaBAkbeAHymKGYowMAClL5KODgiIWERAMSUAGYiGCxoAAowjKxxiaElKkpo6BRoSiJIIq5Y3GgwAJTF2hG+GGhFWiVhKhyMKCLlleg1dQ2KzW19ScAA9Om9YeaLNnYcwX1GeCGclByUKIZp1hzSnCoUci0a81QZNyhydGBIGAAylI1cMCoAORgXz+QF4EQoGDWNAte7tHRgX50booIFZZi5fKscGQ6GwgDcxRwIXwISUqCQZwuVw4KnhjxgFWsMA4IgeYW0GKxBTOSAQ3WJpM0ODmmikVFoDBY7C4PD43JQSI4AHNxJJyJLZApuKoOTo0HpUGhDCZHuYlDRMigBAg0F4QJZGDByjBDAAmVhyUafEE1SzIDxBBERKIxVYldaW6g7bRm0WmDhcADuGuktDUIV8hkBTqUAnBIV0rH0JuMxXMgmErAcHEcIL8jAdSuaaCCABUkRgRB9u1NfjMYOsqzLY5HFcqVWO1iPRE4G-5myCmR2uz3PrV6gOmkPFrPWNOUOsJ0cpwiQnsEn1OIZ9doW0zqZcqNdbgzOSh+c00cC0AWxSUJJ9EBoq4EAA)

**Comment by dagnelies** [comment]
As a comment, using a simple `<div><mycomp on-hello="there">This has a listener</mycomp></div>` will not work either.

**Comment by evs-chris** [comment]
I poked at it a bit, and it looks like there may be a bug with instance contexts immediately inside of a yield - it is finding the `div` from the context above inside the `something` rather than the `mycomp`. I'm not 100% sure though, as I'll have to noodle out the meaning of the nested contexts in the yield, as those can be a bit confusing from the code side.

**Comment by evs-chris** [comment]
This is now fixed in 0.10.4. Thanks for the report!
